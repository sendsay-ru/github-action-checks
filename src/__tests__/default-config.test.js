const { render } = require('../utils');
const source = require('../../config/source');
const defaultConfig = require('../../config/default');

describe('Render by default config', () => {
  it('should expose a function', () => {
    expect(render).toBeDefined();
  });

  it('should return expected value', () => {
    expect(render(source, defaultConfig)).toMatchSnapshot();
  });
});
