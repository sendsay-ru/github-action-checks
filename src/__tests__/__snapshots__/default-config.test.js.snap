// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render by default config should return expected value 1`] = `
[
  {
    "content": "name: Check PR
on: [pull_request]

env:
  NODE_ENV: CI

concurrency:
  group: \${{ github.workflow }}-\${{ github.event.pull_request.number }}
  cancel-in-progress: true

jobs:

  checks:

    runs-on: ubuntu-latest

    outputs:
      eslint: \${{ steps.eslint.outputs.result }}
      prettier: \${{ steps.prettier.outputs.result }}

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Preparing
        uses: ./.github/actions/check-install

      - name: ESLint
        id: eslint
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run eslint

      - name: Prettier
        id: prettier
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run prettify

  tests:

    runs-on: ubuntu-latest

    outputs:
      unit: \${{ steps.unit.outputs.result }}

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Preparing
        uses: ./.github/actions/check-install

      - name: Unit
        id: unit
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run test


",
    "output": ".github/workflows/check-pr.yml",
  },
  {
    "content": "name: Check production
on:
  push:
    branches:
      - stable
      - master
    tags:
      - v**

env:
  NODE_ENV: CI
  FORCE_COLOR: 1

jobs:

  checks:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Preparing
        uses: ./.github/actions/check-install

      - name: ESLint
        id: eslint
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run eslint

      - name: Prettier
        id: prettier
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run prettify

  tests:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Preparing
        uses: ./.github/actions/check-install

      - name: Unit
        id: unit
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run test

",
    "output": ".github/workflows/check-prod.yml",
  },
  {
    "content": "name: Preparing
description: preparing to run the checks

runs:
  using: composite
  steps:
    - name: Setup node
      uses: actions/setup-node@v3
      with:
        node-version: 16

    - name: Install packages
      shell: bash
      run: npm install

",
    "output": ".github/actions/check-install/action.yml",
  },
  {
    "content": "name: Run check
description: run check script

inputs:
  script:
    required: true
    description: script command

outputs:
  result:
    description: "Script result"
    value: \${{ steps.result.outputs.status }}

runs:
  using: composite
  steps:
    - name: Run script
      id: script
      shell: bash
      run: \${{ inputs.script }}

    - name: Results
      id: result
      if: always()
      env:
        SUCCESS_ICON: "☘️"
        FAILURE_ICON: "🐙"
      shell: bash
      run: |
        if [[ \${{ steps.script.outcome }} == 'success' ]]; then echo "status=$SUCCESS_ICON" >> $GITHUB_OUTPUT; else echo "status=$FAILURE_ICON" >> $GITHUB_OUTPUT; fi
",
    "output": ".github/actions/check-run/action.yml",
  },
]
`;
